src/MixpanelEventForwarder.js

```diff
@@ -137,11 +137,50 @@ var constructor = function () {
         }
     }
 
-    function onUserIdentified(user) {
-        var idForMixpanel;
-        var userIdentities = user.getUserIdentities()
+    function getUserIdentities(user) {
+        return user.getUserIdentities()
             ? user.getUserIdentities().userIdentities
             : {};
+    }
+
+    function onLoginComplete(user) {
+        var userIdentities = getUserIdentities(user);
+
+        // When mParticle identifies a user, the user might
+        // actually be anonymous. We only want to send an
+        // identify request to Mixpanel if the user is
+        // actually known. Only known (logged in) users
+        // will have userIdentities.
+        if (!isEmpty(userIdentities)) {
+            sendIdentifyRequest(user, userIdentities);
+        } else {
+            return 'Logged in user does not have user identities and will not be sent to Mixpanel to Identify';
+        }
+    }
+
+    function onLogoutComplete() {
+        if (!isInitialized) {
+            return (
+                'Cannot call logout on forwarder: ' + name + ', not initialized'
+            );
+        }
+
+        try {
+            mixpanel.mparticle.reset();
+
+            return 'Successfully called reset on forwarder: ' + name;
+        } catch (e) {
+            return 'Cannot call reset on forwarder: ' + name + ': ' + e;
+        }
+    }
+
+    function sendIdentifyRequest(user, userIdentities) {
+        var idForMixpanel;
+
+        if (isEmpty(userIdentities)) {
+            userIdentities = getUserIdentities(user);
+        }
```

Comment:
I don't think `userIdentities` is ever passed to `sendIdentifyRequest`, so it's always `null`.

Reply:
I don't think this solves for the use case where an mParticle identify call with 0 user identities is done, returning an anonymous MPID.   mixpanel would still call mixpanel.identify(MPID), and think that this anonymous user is known.  I would recommend creating `onIdentifyComplete`, and having it check for the existence of user identities. If there are 0, do nothing. If there are any, then call `sendIdentifyRequest`.